src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs

```diff
@@ -295,6 +306,38 @@ private static XmlSerializerFormatAttribute ConvertFromServiceModelXmlSerializer
             return xmlSerializerFormatAttribute;
         }
 
+        public static FaultContractAttribute ConvertFromServiceModelFaultContractAttribute(object attr)
+        {
+            Fx.Assert(attr.GetType().FullName.Equals(ServiceReflector.SMFaultContractAttributeFullName), "Expected attribute of type S.SM.FaultContractAttribute");
+            bool hasProtectionLevel = GetProperty<bool>(attr, "HasProtectionLevel");
+            if (hasProtectionLevel)
+            {
+                // ProtectionLevel isn't supported yet so if it was set on the S.SM.SCA, then we can't do the mapping so throw
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new PlatformNotSupportedException("System.ServiceModel.OperationContractAttribute.ProtectionLevel"));
+            }
```

Comment:
I've seen contracts set ProtectionLevel but set it to None. Some people like to explicitly set everything. In that case, it would be safe to do the mapping. Could you modify the logic here to only throw if it was set to anything other than None?

Reply:
The property HasProtectionLevel does not exist in CoreWCF attributes it only exist in SSM ones.
I reproduced the logic that already exist to convert other SSM attributes. please see:
- https://github.com/CoreWCF/CoreWCF/blob/8a15b417fe2f6fedafd88d1c210fb64dad99ae31/src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs#L165
- https://github.com/CoreWCF/CoreWCF/blob/8a15b417fe2f6fedafd88d1c210fb64dad99ae31/src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs#L216
- https://github.com/CoreWCF/CoreWCF/blob/8a15b417fe2f6fedafd88d1c210fb64dad99ae31/src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs#L305

May I open another issue to support explicit `HasProtectionLevel.None` conversion for all SSM attributes in a separate pull request ? it will also cover adding appropriate unit tests.

Reply:
Consistency is more important in this case I believe so I agree with your suggestion. Please open an issue on this and I'll merge this PR.